app/assets/javascripts/core/gtm.js

```diff
@@ -0,0 +1,56 @@
+(function() {
+
+  function filterFormSubmitListner(event) {
+    window.dataLayer = window.dataLayer || [];
+    var form = document.querySelector('form[name="organisation-picker"]');
+
+    var date_range = form.querySelector('input[name="date_range"]:checked').value;
+    window.dataLayer.push({
+      event: 'filter-form',
+      field: 'date_range',
+      value: date_range
+    });
+
+    var search_term = form.querySelector('input[name="search_term"]').value;
+    if(search_term.length > 0) {
+      window.dataLayer.push({
+        event: 'filter-form',
+        field: 'search_term',
+        value: search_term
+      });
+    };
+
+    var document_type = form.querySelector('select[name="document_type"]');
+    window.dataLayer.push({
+      event: 'filter-form',
+      field: 'document_type',
+      value: document_type.selectedOptions[0].value
+    });
+
+    var organisation = form.querySelector('select[name="organisation_id"]');
+    window.dataLayer.push({
+      event: 'filter-form',
+      field: 'organisation',
+      value: organisation.selectedOptions[0].text
+    });
+
+  };
+
+  function init() {
+    document.getElementById("filter-form").addEventListener("submit", filterFormSubmitListner, false);
```

Comment:
Should we check for the element to exist before adding the event listener?

Reply:
Fixed in 42758de